contrib/trace_propagators/build.gradle

```diff
@@ -12,6 +12,9 @@ ext.moduleName = "io.opentelemetry.contrib.trace.propagation"
 dependencies {
     api project(':opentelemetry-api')
 
+    implementation libraries.guava,
+            platform(boms.guava)
```

Comment:
I just realized that this means the B3 and Jaeger propagators now depend on guava. I'm concerned that will be surprising, and potentially lead to version conflicts down the road. 

Opinions, @carlosalberto and @bogdandrutu ?

Reply:
I'm actually only using the `@VisibleForTesting` annotation (which I think is valuable for b3 and jaeger propagators as well). The main package has depended on Guava too. IMO thinking long term the `contrib` will need Guava now or later. It is better to have it before GA

Reply:
My concern is that you're forcing a guava dependency on the B3 and Jaeger propagators when they don't need it (and likely don't want it). I'd rather just replace the annotation with a comment on those methods, rather than add guava here.

Reply:
We were initially working in the sdk_contrib repo and I think there it makes sense to use Guava where helpful since the sdk depends on it anyways. But agree for the API we don't really need it, at least for now.